### PR TITLE
fix: Lint Go code(#12522)

### PR DESCRIPTION
**Pull request by juunini**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

# Description

[ISSUE #12522]
Change `io/util` to `io`

### CODE REVIEW

File: cmd/argocd/commands/applicationset_test.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ SyncPolicy:         Automated
 			printAppSetSummaryTable(tt.appSet)
 			w.Close()
 
-			out, err := ioutil.ReadAll(r)
+			out, err := io.ReadAll(r)
```
Comment (34fathombelow) [inline cmd/argocd/commands/applicationset_test.go:188]:
```suggestion
			out, err := os.ReadAll(r)
```

Comment (34fathombelow) [reply cmd/argocd/commands/applicationset_test.go:188]: Let's keep things a little simpler and just use the `os` package to avoid additional imports.

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/applicationset_test.go (hunk 1 above)
Comment (34fathombelow) [inline cmd/argocd/commands/applicationset_test.go:188]:
```suggestion
			out, err := os.ReadAll(r)
```

Comment (34fathombelow) [reply cmd/argocd/commands/applicationset_test.go:188]: Let's keep things a little simpler to avoid additional imports.

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/applicationset_test.go (hunk 1 above)
Comment (34fathombelow) [inline cmd/argocd/commands/applicationset_test.go:188]:
```suggestion
			out, err := os.ReadAll(r)
```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]
Let's keep things a little simpler to avoid additional imports.

**Comment by juunini** [comment]
Screenshot_20230219_012100_Chrome

@34fathombelow `os.ReadAll` is not exists

File: cmd/argocd/commands/applicationset_test.go (hunk 1 above)
Comment (juunini) [reply cmd/argocd/commands/applicationset_test.go:188]: image

I was change `io.ReadAll` to `os.ReadAll` and remove import `io`.  
But, `os.ReadAll` is not exists.  

so, git reset and force push before commit

**Review by 34fathombelow — APPROVED** [review]
lgtm!  Thanks so much for your contribution

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**